resolver/resolver.go

```diff
@@ -34,29 +42,25 @@ func (e ErrNoLink) Error() string {
 	return fmt.Sprintf("no link named %q under %s", e.Name, e.Node.String())
 }
 
-// ResolveOnce resolves path through a single node
-type ResolveOnce func(ctx context.Context, ds ipld.NodeGetter, nd ipld.Node, names []string) (*ipld.Link, []string, error)
-
 // Resolver provides path resolution to IPFS
-// It has a pointer to a DAGService, which is uses to resolve nodes.
+// It has a pointer to a FetcherConfig, which is uses to resolve nodes.
 // TODO: now that this is more modular, try to unify this code with the
 //       the resolvers in namesys
 type Resolver struct {
-	DAG ipld.NodeGetter
-
-	ResolveOnce ResolveOnce
+	FetchConfig fetcher.FetcherConfig
```

Comment:
I'd vote weakly (both weak vote and weakly held position) that that might be premature abstraction.  If there's probably only that one shape of config fields in the fetcher component, we might as well admit it, because it'll be more discoverable and easier to autocomplete.  But, I don't have strong feelings about this, not having a super complete grasp of how diverse fetcher config might be planned to become.

Reply:
it ended up being an interface anyway! (due to various other changes happening in go-fetcher) :)